batchSystems/gridengine.py

```diff
@@ -69,7 +71,7 @@ def prepareQsub(cpu, mem):
                      "LD_LIBRARY_PATH=%s" % os.environ["LD_LIBRARY_PATH"]]
     reqline = list()
     if cpu is not None:
-        reqline.append("p="+str(cpu))
+        qsubline.extend(["-pe", "shm", str(int(cpu))])
```

Comment:
Hi Hannes,

I don't think "-pe shm" and "-pe smp" are equivalent, but I'm not entirely
sure.

For example, on the computer I use, the results of the command qconf -spl
are:
orte
shm

So I don't think I can use the smp parallel environment on this particular
computer without making some changes.

I can't think of an elegant solution at the moment, but maybe you could
prompt the user for the name of the appropriate parallel environment, just
once, during package setup.

cheers,
Tom

On Tue, Oct 6, 2015 at 6:05 PM, Hannes Schmidt <notifications@github.com>
wrote:

> In batchSystems/gridengine.py
> <https://github.com/benedictpaten/jobTree/pull/41#discussion_r41341883>:
>
> > @@ -69,7 +71,7 @@ def prepareQsub(cpu, mem):
> >                       "LD_LIBRARY_PATH=%s" % os.environ["LD_LIBRARY_PATH"]]
> >      reqline = list()
> >      if cpu is not None:
> > -        reqline.append("p="+str(cpu))
> > +        qsubline.extend(["-pe", "shm", str(int(cpu))])
>
> Toil already switched to -pe but it uses -pe smp instead of -pe shm. Do
> you think they are equivalent?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/benedictpaten/jobTree/pull/41/files#r41341883>.
>